cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +354,155 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
+
+
+@pytest.fixture(scope='module')
+def setup(domain, klass, namespace):
+    """This fixture creates common domain, namespace, two classes, instance and method. This setup
+    is common for both parameterized tests"""
+
+    state1 = fauxfactory.gen_alpha()
+    state2 = fauxfactory.gen_alpha()
+    state3 = fauxfactory.gen_alpha()
+
+    klass.schema.add_fields(*[{'name': state, 'type': 'State'}
+                              for state in [state1, state2, state3]])
+
+    # Adding method to execute via parent instance(state machine instance)
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script="""
+               \n$evm.log(:info, "Hello from method of parent instance")
+                \nexit MIQ_STOP
+                """
+    )
+
+    # Adding parent instance to call automate method and child instances in other class
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={state3: {"value": f"METHOD::{method.name}"}}
+    )
+
+    # Creating other class for two child instances
+    klass2 = namespace.classes.create(
+        name=fauxfactory.gen_alpha(),
+        display_name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha()
+    )
+    klass2.schema.add_fields(*[{'name': state, 'type': 'State'}
+                               for state in [state1, state2, state3]])
+
+    yield klass2, instance, state1, state2, state3
+    method.delete_if_exists()
+    instance.delete_if_exists()
+    klass2.delete_if_exists()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1441353])
+@pytest.mark.parametrize("process", ["miq_stop", "miq_abort"])
+def test_miq_stop_abort_with_state_machines(request, setup, process, domain, klass, namespace):
+    """
+    Bugzilla:
+        1441353
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+        setup:
+            1. Create automate domain, namespace, two classes - class A and class B
+            2. Class A and B should have schema of type - State
+            3. Create instance A1 and method A1 under class A. Here instance A1 should call method
+               A1
+            4. Create two instances B1, B2 and three methods B1, B2, B3 under class B. Here instance
+               B1 should call two methods B1 and B2. Also instance B2 should call method B3
+            5. Now Update instance A1 to call instance B1 and B2
+        testSteps:
+            1. Navigate to Automation > Automate > Simulation page and execute instance A1
+            2. If all the methods contain process - MIQ_STOP
```

Comment:
Can you modify this test step a little? I think it should be more like - _" Execute all the methods containing process - MIQ_STOP."_
I think this would make more sense, let me know what you think.
Same for step3.

Reply:
@valaparthvi 
- So if you see here ```step 1 : Navigate to Automation > Automate > Simulation page and execute instance A1``` is already indicates that you need to only execute that instance(A1).
- So step 2 and 3 are for understanding to check logs. Hence keeping as it is.
- Actually your sentence does not satisfy test case. Because it doesn't mean we are executing methods where all methods are containing ```MIQ_STOP``` or ```MIQ_ABORT```